### PR TITLE
feat: allow specified fields in subgraph errors

### PR DESCRIPTION
**Pull request by thisisnithin**
Related PR: https://github.com/wundergraph/graphql-go-tools/pull/912

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]

File: router/pkg/config/config.schema.json [hunk 1]
```diff
@@ -1860,6 +1860,11 @@
           "default": false,
           "description": "Omit the extensions field of Subgraph errors. If the value is true (default: false), the extensions field of Subgraph errors will be omitted. This is useful in case you want to avoid leaking internal information to the client. Some users of GraphQL leverage the errors.extensions.code field to implement error handling logic in the client, in which case you might want to set this to false."
         },
+        "omit_custom_fields": {
+          "type": "boolean",
+          "default": false,
+          "description": "Omit custom field of Subgraph errors in pass through mode. If the value is true (default: false), all fields except message and extensions will be omitted."
```
Comment (StarpTech) [inline router/pkg/config/config.schema.json:1866]: What about locations field? Is it still propagated when set `omit_locations` false?

File: router/pkg/config/config.schema.json [hunk 2]
```diff
@@ -1860,6 +1860,11 @@
           "default": false,
           "description": "Omit the extensions field of Subgraph errors. If the value is true (default: false), the extensions field of Subgraph errors will be omitted. This is useful in case you want to avoid leaking internal information to the client. Some users of GraphQL leverage the errors.extensions.code field to implement error handling logic in the client, in which case you might want to set this to false."
         },
+        "omit_custom_fields": {
+          "type": "boolean",
+          "default": false,
```
Comment (StarpTech) [inline router/pkg/config/config.schema.json:1865]: Default should be true to not leak information.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/pkg/config/config.schema.json (hunk 1 above)
Comment (thisisnithin) [reply router/pkg/config/config.schema.json]: Using allowlist instead

**Review by StarpTech — APPROVED** [review]
LGTM

File: router-tests/error_handling_test.go [hunk 3]
```diff
@@ -601,6 +601,7 @@ func TestErrorPropagation(t *testing.T) {
 				cfg.Mode = config.SubgraphErrorPropagationModePassthrough
```
Comment (StarpTech) [inline router-tests/error_handling_test.go:601]: Please add a test that validates the functionality of the new flag.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router-tests/error_handling_test.go (hunk 3 above)
Comment (thisisnithin) [reply router-tests/error_handling_test.go:601]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**